### ISSUE TITLE
Querier fails to start until store-gateway is up and running

### ISSUE THREAD

**Issue by amckinley** [cortexproject/cortex#2991]
While building out a new Cortex cluster on v1.3.0-rc0, I noticed that when the store-gateway pods fail to start (for unrelated reasons), the queriers also fail to start and get stuck in a crash loop with logs similar to these: https://pastebin.com/Dnet1FUr

If this is the desired behavior, you can go ahead and close this, but imo it would be better to just keep retrying until the store-gateway pods are up and running.

**Comment by pracucci** [comment]
I don't think the error you got in the querier is related to the store-gateway. Queriers should start even if store-gateways are disrupted (then queries will fail at execution time, but that's expected because at least 1 store-gateway is required).

You got this error in the querier:
```
level=error ts=2020-08-06T20:23:30.893013454Z caller=cortex.go:319 msg="module failed" module=querier err="failed to start querier, because it depends on module store-queryable, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: unable to start blocks storage queryable subservices: not healthy"
```

Which mentions that the `store-queryable` (which is a querier internal component) has failed to start, because:
```
unable to start blocks storage queryable subservices
```

The real issue is that the root cause of why one of the store-queryable failed to start is missing in the logs (needs to be fixed).

Have you the blocks sharding enabled in the store-gateway? If so, could be possible that the ring backend (eg. consul) was unhealthy? It could be one of the possible reasons why the `store-queryable` could fail to start.

**Comment by pracucci** [comment]
Actually, after looking at the code, I believe the only case this could happen is if the querier fails to scan the object store (bucket) for the initial blocks scanning operation.

**Comment by amckinley** [comment]
> Have you the blocks sharding enabled in the store-gateway? If so, could be possible that the ring backend (eg. consul) was unhealthy? It could be one of the possible reasons why the store-queryable could fail to start.

Yeah, we were in the process of bringing up a new cluster when we encountered this error. I can't remember the exact timeline, but at different points we had problems getting consul running and with making the S3 buckets storing the TSDB accessible to the pods. I saw the error message referencing `store-queryable` and assumed it was a reference to the `store-gateway` pods, which were stuck in a crash loop at the time we saw those logs.
